src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,122 @@
+package duke.parser;
+
+import duke.DukeException;
+import duke.commands.*;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.ToDo;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a parser to extract information from user input.
+ */
+public class Parser {
+
+    /**
+     * Parses user input into commands.
+     *
+     * @param input The user input string.
+     * @return The command corresponding to the input.
+     * @throws DukeException If the input string have errors in its format.
+     */
+    public static Command parse(String input) throws DukeException {
```

Comment:
I like how you have created separate functions for each case to keep your method short!

Good understanding of switch case is displayed here (i.e no unnecessary break)